src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -616,3 +617,37 @@
                                  (icons/icon {} (if (:collapsed? @state) :expand :collapse)))
           body]
          body)))})
+
+(react/defc QueueStatus
+  {:render
+   (fn [{:keys [state this]}]
+     (let [{:keys [queue-status queue-error]} @state
+           {:keys [queue-time queue-position queued active]} queue-status]
+       [:div {:style {:marginBottom "0.5em"}}
+        (cond
+          queue-error (style/create-server-error-message queue-error)
+          (not queue-status) [Spinner {:text "Loading submission queue status..."}]
```

Comment:
UX nitpick: when refreshing, the component shrinks vertically, causing the entire layout to jitter. Could you fix the spinner to the same vertical height as the main display?